docs/guides/02_writing_commands.md

```diff
@@ -0,0 +1,233 @@
+
+    private readonly IRepository<Project, SId<Project>> projects;
+
+    public CreateProjectCH(IRepository<Project, SId<Project>> projects)
+    {
+        this.projects = projects;
+    }
+
+    public Task ExecuteAsync(CoreContext context, CreateProject command)
+    {
+        var project = Project.Create(command.Name);
+        projects.Add(project);
+
+        logger.Information("Project {ProjectId} added", project.Id);
+    }
+}
+```
+
+> **Warning :warning:**
+> According to DDD rules, a single command should only modify a single aggregate as an aggregate should be a transaction boundary in the system. If there is a need to modify multiple aggregates as a part of a single operation, raise events and modify all aggregates one by one in event handlers.
+
+The logic of the handler is straightforward - a new project with a given name is created and added to `projects` repository. The `IRepository` interface is a part of Core Library which allows to abstract the logic of persisting domain entities. A repository (and other services) should be injected into a command handler using dependency injection. Also, notice that the `Add` method used to persist newly created project is synchronous. The actual asynchronous operation in the database is performed by `StoreAndPublishEvents()` element in the pipeline, see [pipeline](./0X_pipeline.md) for more info. A basic repository for projects allowing for operations on database may look like this:
+
+```csharp
+public class ProjectsRepository<TContext>
+    : EFRepository<Purchase, SId<Purchase>, TContext>
+    where TContext : DbContext
+{
+    public ProjectsRepository(TContext dbContext)
+        : base(dbContext) { }
+
+    public override Task<Purchase?> FindAsync(SId<Purchase> id, CancellationToken cancellationToken = default)
+    {
+        return DbSet.AsTracking().FirstOrDefaultAsync(p => p.Id == id, cancellationToken)!;
+    }
+}
+```
+
+Here `EFRepository` is a class from Core Library containing default implementations for `Add`, `Delete` and `Update` methods which can be used to alter the contents of the database. Usually `FindAsync` method can be implemented as a simple retrieval of an entity based on Id but more complicated operations can also be done in the method if they are needed.
+
+> **Warning :warning:**
+> If an implemented command does not work, make sure that all referenced types are properly registered in the dependency injection container.
+
+## More complex commands
+
+The command for creating a project is a really basic one - it only takes a name for new project and all of the validation can be done without checking the state of the domain. We will now show a slightly more complex command which will be responsible for adding tasks to existing projects.
+
+```csharp
+[AllowUnauthorized]
+public class AddTasksToProject : ICommand
+{
+    public string ProjectId { get; set; }
+    public List<TaskDTO> Tasks { get; set; }
+
+    public static class ErrorCodes
```

Comment:
Shouldn't we include `TaskDTO.ErrorCodes` here?

Reply:
I am not sure, I don't include such error codes in the command's error codes. Is it a mistake?

Reply:
Not really a mistake, but having that might help clients (they will have all error codes in a single place then).